### PR TITLE
[Enhancement] add keepExistingConfig option to addDataToMap

### PR DESCRIPTION
**Pull request by heshan0131**
- Keep old visState config if `keepExistingConfig` is true
- [Breaking] Remove `updateVisDataComposed` updater, everything goes though `addDataToMap`
- [Breaking] `options.centerMap` in `addDataToMap` overrides `config.mapState`
Signed-off-by: Shan He <heshan0131@gmail.com>

### CODE REVIEW

**Comment by heshan0131** [comment]
https://github.com/keplergl/kepler.gl/issues/343

File: docs/api-reference/reducers/vis-state.md [hunk 1]
```diff
@@ -499,7 +503,7 @@ Add new dataset to `visState`, with option to load a map config along with the d
             -   `action.datasets.data.fields` **[Array][68]&lt;[Object][63]>** **\*required** Array of fields,
                 -   `action.datasets.data.fields.name` **[string][64]** **\*required** Name of the field,
             -   `action.datasets.data.rows` **[Array][68]&lt;[Array][68]>** **\*required** Array of rows, in a tabular format with `fields` and `rows`
-    -   `action.options` **[Object][63]** option object `{centerMap: true}`
+    -   `action.options` **[Object][63]** option object `{centerMap: true, keepExistingConfig: false}`
```
Comment (macrigiuseppe) [inline docs/api-reference/reducers/vis-state.md]: should readOnly go here too?

File: src/reducers/vis-state-updaters.js [hunk 2]
```diff
@@ -1086,22 +1090,23 @@ export const toggleLayerForMapUpdater = (state, action) => {
  * @param {Array<Object>} action.datasets.data.fields - ***required** Array of fields,
  * @param {string} action.datasets.data.fields.name - ***required** Name of the field,
  * @param {Array<Array>} action.datasets.data.rows - ***required** Array of rows, in a tabular format with `fields` and `rows`
- * @param {Object} action.options option object `{centerMap: true}`
+ * @param {Object} action.options option object `{centerMap: true, keepExistingConfig: false}`
```
Comment (macrigiuseppe) [inline src/reducers/vis-state-updaters.js]: readOnly?

File: examples/demo-app/src/app.js [hunk 3]
```diff
@@ -48,7 +48,8 @@ const KeplerGl = require('kepler.gl/components').injectComponents([
 import sampleTripData from './data/sample-trip-data';
 import sampleGeojson from './data/sample-small-geojson';
 import sampleGeojsonPoints from './data/sample-geojson-points';
-import sampleH3Data from './data/sample-hex-id-csv';
+import sampleGeojsonConfig from './data/sample-geojson-config';
```
Comment (macrigiuseppe) [inline examples/demo-app/src/app.js]: this file doesn't exist or your forgot to push it to the repo. That's why the netlify build is failing

File: src/reducers/combined-updaters.js [hunk 4]
```diff
@@ -73,66 +73,12 @@ import KeplerGlSchema from 'schemas';
...
- * @param {string} action.datasets.data.fields.name - ***required** Name of the field,
- * @param {Array<Array>} action.datasets.data.rows - ***required** Array of rows, in a tabular format with `fields` and `rows`
- * @param {Object} action.options
- * @param {Boolean} action.options.centerMap
- * @param {Boolean} action.options.readOnly
- * @param {Object} action.config
- * @returns {Object} nextState
- * @public
- */
-export const updateVisDataUpdater = (state, action) => {
```
Comment (heshan0131) [inline src/reducers/combined-updaters.js]: removed, everything goes through `addDataToMap`

**Comment by heshan0131** [comment]
@macrigiuseppe please take another look, made some breaking changes to cleanup logic

**Pull request merged**